python/ray/tune/trial.py

```diff
@@ -224,18 +224,33 @@ def __init__(self,
         if trainable_cls:
             default_resources = trainable_cls.default_resource_request(
                 self.config)
-            if default_resources:
-                if resources:
-                    raise ValueError(
-                        "Resources for {} have been automatically set to {} "
-                        "by its `default_resource_request()` method. Please "
-                        "clear the `resources_per_trial` option.".format(
-                            trainable_cls, default_resources))
+
+            # If Trainable returns resources, do not allow manual overrid via
+            # `resources_per_trial` by the user.
+            if default_resources and (resources or placement_group_factory):
+                raise ValueError(
+                    "Resources for {} have been automatically set to {} "
+                    "by its `default_resource_request()` method. Please "
+                    "clear the `resources_per_trial` option.".format(
+                        trainable_cls, default_resources))
+
+            # New way: Trainable returns a PlacementGroupFactory object.
+            if isinstance(default_resources, PlacementGroupFactory):
+                placement_group_factory = default_resources
+                resources = None
+            # Set placement group factory to None for backwards compatibility.
+            else:
+                placement_group_factory = None
                 resources = default_resources
+
         self.location = Location()
 
         self.resources = resources or Resources(cpu=1, gpu=0)
-        self.placement_group_factory = placement_group_factory
+        if isinstance(resources, PlacementGroupFactory):
+            self.placement_group_factory = resources
+        else:
+            self.placement_group_factory = placement_group_factory
+
```

Comment:
```suggestion
        self.placement_group_factory = placement_group_factory
```

Reply:
We don't need this block anymore (`placement_group_factory` is an argument of the constructor). With the changes above `resources` will never hold anything other than `None` or a `Resources` object

Reply:
removed.